### PR TITLE
test: Fix TestActivePages races and bugs

### PR DESCRIPTION
**Pull request by martinpitt**
With full chromium instead of chromium-headless, the `focus(".terminal")`
focused *away from* the actual element that receives the input, so that
typing the sleep command went into Nirvana. Just drop that, it's
unnecessary -- we want the Terminal page to automatically focus the
input. (check-system-terminal also does not focus explicitly).

Drop the weird `line_text()` helper, and wait until the terminal shows a
full prompt (with '`$`), instead of just "anything". This does the same
as check-system-terminal (albeit a bit simplified).

Wait for the default selection of the "delete frame" dialog to happen.
Previously the clickery started too early before the dialog got fully
initialized, and then missed to actually stop the terminal frame.

Fix the waiting for "sleep command goes away"; before it waited for the
command to be present, thus was a no-op.

Replace the broken `ps | grep` with `pgrep -x`, which avoids catching
itself. Despite the effort with using a regexp, the grep command still
found itself. Also, `sleep` is a bad program to use, as on Debian/Ubuntu
that just gets reparented to pid 1 after its parent (the terminal shell)
dies, so it never gets killed (this was hidden by the previous bug where
the second wait loop was the wrong way around). Replace this with `cat`,
which will properly exit on SIGPIPE, and rename it to `kitten` to
ensure that our pgrep does not catch some unrelated cat.

Use a single shell commmand to the VM instead of a heavy polling loop.

--- 

Broken out from #15378, as this test case was so broken that it needs a *lot* of attention. It's still not 100% right.

### CODE REVIEW

**Comment by martinpitt** [comment]
[This](https://logs.cockpit-project.org/logs/pull-15378-20210219-144940-b0b23039-fedora-33/log.html#325) is the remaining flake that I'm hunting now. I can reproduce this locally reasonably well.

**Comment by martinpitt** [comment]
I fixed another two race conditions. Now this test finally survived 2x5 parallel runs, once with debian-testing and once with fedora-33. Let's see whether the bots finally agree.

**Comment by marusak** [comment]
[coreos](https://logs.cockpit-project.org/logs/pull-15385-20210219-183716-c5caf340-fedora-coreos/log.html#325) does not agree

**Comment by martinpitt** [comment]
@marusak : Yep, I know, I just didn't get  to that yesterday any more. Should be fixed now, I ran the test several times locally on coreos.

File: test/verify/check-shell-active-pages [hunk 1]
```diff
@@ -19,28 +19,23 @@
...
 
     def testBasic(self):
         b = self.browser
         m = self.machine
 
-        self.login_and_go("/system")
+        # disable preloads; they are different between OSes, make the counting hard, and happen asynchronously
+        # disable_preload(), sed_file() etc. need writable file system
+        for p in ["packagekit", "playground", "systemd"]:
+            self.restore_dir("/usr/share/cockpit/" + p)
```
Comment (martinpitt) [inline test/verify/check-shell-active-pages:35]: I was just thinking, this should actually go into the implementation of `disable_preload()`. We don't use it anywhere on OSTree until now. I'll wait for a round of test results and then update that.

Comment (martinpitt) [reply test/verify/check-shell-active-pages]: Previous round's results all succeeded everywhere, pushed this cleanup.

File: test/verify/check-shell-active-pages [hunk 2]
```diff
@@ -69,36 +69,38 @@ class TestActivePages(MachineCase):
...
-
         # wait for prompt in first line
-        b.wait_text_not(line_sel(1), line_text(""))
+        b.wait_visible(".terminal .xterm-accessibility-tree")
+        b.wait_in_text(line_sel(1), '$')
 
-        # run the sleep command
-        b.key_press("sleep 999999\r")
+        # run a command that we can easily identify, and which will die with the terminal
+        b.key_press("bash -c 'exec -a kitten cat'\r")
```
Comment (marusak) [inline test/verify/check-shell-active-pages:72]: kitten - cat :smile:

**Review by marusak — APPROVED** [review]
Very nice job! It simplifies it and also makes it more robust.

**Pull request merged**